actions/drain_node_handler.go

```diff
@@ -86,9 +86,20 @@ func (h *drainNodeHandler) Handle(ctx context.Context, data interface{}) error {
 		// If force is set and evict timeout exceeded delete pods.
 		deleteCtx, deleteCancel := context.WithTimeout(ctx, h.cfg.podsDeleteTimeout)
 		defer deleteCancel()
-		if err := h.deleteNodePods(deleteCtx, log, node); err != nil {
+		err = h.deleteNodePods(deleteCtx, log, node, metav1.DeleteOptions{})
+		if err == nil {
+			log.Info("node drained")
+			return nil
+		}
+		if !errors.Is(err, context.DeadlineExceeded) {
 			return fmt.Errorf("forcefully deleting pods: %w", err)
 		}
+
+		deleteForceCtx, deleteForceCancel := context.WithTimeout(ctx, h.cfg.podsDeleteTimeout)
+		defer deleteForceCancel()
+		if err = h.deleteNodePods(deleteForceCtx, log, node, *metav1.NewDeleteOptions(0)); err != nil {
+			return fmt.Errorf("deleting gracePeriod=0 pods: %w", err)
+		}
```

Comment:
suggestion, maybe this will be more clear: 🤔 
```go

// Try deleteing pods gracefully first, then delete with 0 grace period.
		options := []metav1.DeleteOptions{
			metav1.DeleteOptions{},
			*metav1.NewDeleteOptions(0)
		}

		var err error
		for _, o := options {
			deleteCtx, deleteCancel := context.WithTimeout(ctx, h.cfg.podsDeleteTimeout)
			defer deleteCancel()

			err = h.deleteNodePods(deleteCtx, log, node, o);
			if err == nil {
				return nil
			}
			if !errors.Is(err, context.DeadlineExceeded) {
				return fmt.Errorf("forcefully deleting pods: %w", err)
			}
		}

		return err
```